### ISSUE TITLE
Exclude +/- line markers when copying from diffs

### ISSUE THREAD

**Issue by trevorsmith** [gitx/gitx#43]
I sometimes copy text directly from diffs to paste in other places. It would be amazing if the copied text didn't include the +/- markers at the beginning of each line.

**Comment by andreif** [comment]
Hm, it needs to be an option, since user may want to copy the diff as diff.

**Comment by trevorsmith** [comment]
Maybe we could add one or the other as a context menu item? "Copy as Plain Text" or "Copy as Diff" perhaps.

**Comment by andreif** [comment]
Yeah, something like that. Both of them are plain text though, so maybe some other phrasing could help the user 😄

**Comment by rjmunro** [comment]
Could we use a heuristic like include the + and - if the selection includes lines of both types, or would that just get really annoying?

**Comment by andreif** [comment]
Explicit choice is better I think. And simpler.

**Comment by trevorsmith** [comment]
I agree. If the heuristic fails even 1% of the time it would be confusing and frustrating.

As for the label, I haven been able to think of anything better than "plain text" (which as you pointed out makes no sense). "Copy Lines"? "Copy Code"? Or maybe we make the default copy exclude the diff markers, and then use "Copy Patch" or "Copy Diff"?

**Comment by andreif** [comment]
* I like "Copy Lines" 👍 
* GitX may be used not only for source code, but also Mardown, txt, license, so "Copy Code" won't always fit. 
* The default should include diff since users will expect to copy what thay see selected.

**Comment by rjmunro** [comment]
What might be better is a way to jump from a diff directly to either the before or after version on the file, which you could then copy and paste from freely, or perhaps even save as a whole from the file menu.

**Comment by andreif** [comment]
@rjmunro Split diff as on GitHub?

**Comment by trevorsmith** [comment]
I do like this idea as an additional feature. For this use case, however - manually cherry picking a few random lines - having to jump to another file is even more cumbersome than removing the +/- from the beginning of the lines.

Jumping to a file at a specific point in history would be incredible helpful in other situations, though. Both would be fairly easy to implement.

**Comment by trevorsmith** [comment]
I just discovered that the Stage view already implements a Copy button that copies lines without the diff markers.

**Comment by andreif** [comment]
@trevorsmith Huh, it should probably include the markers though :)

**Comment by trevorsmith** [comment]
How often are you needing to copy lines with the markers, and where are you using them? It's currently not possible to copy the entire patch from the stage view, and very difficult to do from the history view.

**Comment by andreif** [comment]
JetBrains IDEs don't show +/-, just green/red background, and therefore avoid any confusion. 
GitX could do the same, and also add en extra menu option to "Copy As Diff" so that everybody is happy.

<img src="http://i.imgur.com/a4YwqJV.png" width=250>

Though, I feel that it would be a less gitty way.

**Comment by trevorsmith** [comment]
I like that solution. It is less bitty, but using a GUI is arguably not very gritty either. So my vote is for the markerless diffs, but I'm open to whatever the community thinks is best.

I didn't realize most of these views were implemented in JS, so it is far easier for me to contribute. I'm happy give this a shot once we reach a consensus.
